### ISSUE TITLE
RustDesk not sending key strokes from ChromeOS machine

### ISSUE THREAD

**Issue by MattTheQuick** [rustdesk/rustdesk#730]
**Describe the bug you encountered:**
Connecting from Chromebook (with Linux development environment installed) to remote machine allows mouse control but does not send keystrokes. 

**What did you expect to happen instead?**
Typing on local machine does not send keystrokes to remote machine.

**How did you install `RustDesk`?**

Installed .deb to Linux development environment of Chromebook

**RustDesk version and environment**
1.1.9

Samsung Chromebook 4, stock ChromeOS (latest version) with Linux Development Environment (which I think is Debian 10) installed. RustDesk is installed into that partition. RustDesk opens as expected, connects to machine and accepts password for the connection to complete, but typing anything into the remote computer is impossible.

**Comment by blueice777** [comment]
try this:

type **about:config** in address bar, then search for privacy.resistFingerprinting and if it's enabled (**true**), change to **false**.

**Comment by DaSpawn** [comment]
Hi, I just ran into this issue when I upgraded my Rustdesk from 1.1.8 (which worked fine but had secure error) to 1.1.9 and I can no longer type into the machine (but I can send ctrl+alt+del without issue) and the mouse works fine.  

There is a tag of help wanted, is there some way I can help?

**Comment by rustdesk** [comment]
> help wanted

debug to figure out the problem if possible

**Comment by rustdesk** [comment]
we do have big change from 1.1.8 to 1.1.9 to support international keyboards, though still  buggy.

**Comment by DaSpawn** [comment]
I can try to take a look if you can point me in the right direction (I noticed there is a lot of changes between 1.1.8 and 1.1.9). what areas/modules would be best to focus on?

**Comment by rustdesk** [comment]
in 1.1.8, we always send 'A' -> 'Z' char (US keyboard), in 1.1.9, we send unicode. 
For Linux, we use pynput https://github.com/rustdesk/rustdesk/blob/master/libs/enigo/src/linux.rs#L487 via ipc for input simulation.

**Comment by DaSpawn** [comment]
I am digging into, but I just seen this: 

[2022-06-24T16:22:22Z INFO  enigo::linux] pynput service: ./pynput_service.py
[2022-06-24T16:22:22Z WARN  enigo::linux] Failed to connect to /tmp/RustDesk/pynput_service: Connection refused (os error 111)
INFO:TIS: enter
[2022-06-24T16:22:23Z INFO  enigo::linux] Conntected to pynput server
Got pynput connection

I will keep digging

**Comment by rustdesk** [comment]
here is a PR related to pynput, not sure if can help, https://github.com/rustdesk/rustdesk/pull/850

**Comment by funghetto** [comment]
ChromeOS has support for both wayland and X11 apps but as RustDesk can work partially on wayland this is what gets started by default, as a workaround you can start RustDesk with:
`sommelier -X rustdesk `
to ask sommelier to do X11 forwarding for rustdesk, this makes the keyboard work.

**Comment by rustdesk** [comment]
try out nightly build

**Comment by lihanchen** [comment]
Hi, I tried the latest nightly on both client and server, the issue is still there.
sommelier works well.
I am having the latest chromeos with latest Rustdesk client deb for x64

**Comment by kal10bach** [comment]
hi, the same with my chromeos (actual)
- mouse works
- tab, enter, etc. work
- but no characters
thx georg

**Comment by kb9gxk** [comment]
I can confirm the keyboard does not work on a Pixelbook Go.

Modifying the shortcuts with `sommelier -X rustdesk` does fix the issue.

**Comment by macccorehenni** [comment]
It hasn't worked for me

**Comment by remklov** [comment]
We had the same issue on chromebooks. The `sommelier -X rustdesk` fixed the problem. 
To run rustdesk with the ChromeOS Launcher with sommelier you can do this:

From within your `penguin` terminal, edit `/usr/share/applications/rustdesk.desktop` as root and change the line `Exec=rustdesk` to `Exec=sommelier -X rustdesk`.

**Comment by droidgren** [comment]
If the GUI gets very small with sommelier, just add the scale parameter to 0.5 like this: 

`sommelier -X --scale=0.5 --dpi rustdesk`

**Comment by macccorehenni** [comment]
To be clear, this is running on Crostini, not bare metal, and is not an issue of scale, but remote input not being sent.

Cameron Reape

**Comment by droidgren** [comment]
> To be clear, this is running on Crostini, not bare metal, and is not an issue of scale, but remote input not being sent. Cameron Reape
> […](#)
> On Mar 2, 2024 at 13:29 -0800, droidgren ***@***.***>, wrote: If the GUI gets very small with sommelier, just add the scale parameter to 0.5 like this: sommelier -X --scale=0.5 --dpi rustdesk — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>
Yes, but a negative side-effect of running Crostini with the sommelier option was the scale beeing wrong ( at least for me it was ). This fixes it.
